app/models/post.rb

```diff
@@ -10,18 +10,45 @@ class Post < Crowdblog::Post
   searchable do
     text :title, :body, :author_name, :category_names
     string :state
-  end
-
-  def self.grouped_for_archive
-    published_and_ordered.group_by {|p| p.published_at.year }.
-        inject({}) { |mem, value| mem[value[0]] = value[1].
-            group_by {|p| p.published_at.strftime("%B")}; mem }
+    integer :author_id
+    integer :category_ids, multiple: true
+    time :published_at
   end
 
   def category_names
     categories.pluck(:name).join(' ')
   end
 
+  def self.query(query)
+    dates = self.date_range(query)
+    Post.search do
+      fulltext query[:q]
+      with(:author_id, query[:author].to_i) if query[:author].present?
+      with(:category_ids, [query[:category].to_i]) if query[:category].present?
+      with(:state, 'published')
+      with(:published_at, dates)
+      order_by(:published_at, :desc)
+      paginate page: query[:page] || 1, per_page: query[:per_page] || 30
+    end
+  end
+
+  def self.date_range(params)
+    from = params[:from].blank? ? '2010-01-01' : params[:from]
```

Comment:
@zalapeach I would suggest to move the "2010-01-01" into a Constant variable and then you can use it within the spec test for this model and only declare that variable in one place